go/adbc/driver/bigquery/connection.go

```diff
@@ -0,0 +1,868 @@
+		nestedFields := make([]arrow.Field, 0)
+		for _, record := range strings.Split(fieldRecords, ",") {
+			fieldRecord := strings.TrimSpace(record)
+			recordParts := strings.SplitN(fieldRecord, " ", 2)
+			if len(recordParts) != 2 {
+				return arrow.Field{}, adbc.Error{
+					Code: adbc.StatusInvalidData,
+					Msg:  fmt.Sprintf("invalid field record `%s` for type `%s`", fieldRecord, dataType),
+				}
+			}
+			fieldName := recordParts[0]
+			fieldType := recordParts[1]
+			nestedField, err := buildSchemaField(fieldName, fieldType)
+			if err != nil {
+				return arrow.Field{}, err
+			}
+			nestedFields = append(nestedFields, nestedField)
+		}
+		structType := arrow.StructOf(nestedFields...)
+		if structType == nil {
+			return arrow.Field{}, adbc.Error{
+				Code: adbc.StatusInvalidArgument,
+				Msg:  fmt.Sprintf("Cannot create a struct schema for record `%s`", fieldRecords),
+			}
+		}
+		field.Type = structType
+	default:
+		return arrow.Field{}, adbc.Error{
+			Code: adbc.StatusInvalidArgument,
+			Msg:  fmt.Sprintf("Cannot handle data type `%s`, type=`%s`", dataType, typeString),
+		}
+	}
+	return field, nil
+}
+
+func parsePrecisionAndScale(name, typeString string) (int32, int32, error) {
+	typeString = strings.TrimSpace(typeString)
+	if len(typeString) == 0 {
+		return 0, 0, adbc.Error{
+			Code: adbc.StatusInvalidData,
+			Msg:  fmt.Sprintf("Cannot parse data type `%s` for field `%s`", typeString, name),
+		}
+	}
+
+	if typeString == "NUMERIC" {
+		return 38, 9, nil
+	} else if typeString == "BIGNUMERIC" {
+		return 76, 38, nil
+	}
+
+	values := strings.Split(strings.TrimRight(typeString[strings.Index(typeString, "(")+1:], ")"), ",")
+	if len(values) != 2 {
+		return 0, 0, adbc.Error{
+			Code: adbc.StatusInvalidData,
+			Msg:  fmt.Sprintf("Cannot parse data type `%s` for field `%s`", typeString, name),
+		}
+	}
+
+	precision, err := strconv.ParseInt(values[0], 10, 32)
+	if err != nil {
+		return 0, 0, adbc.Error{
+			Code: adbc.StatusInvalidData,
+			Msg:  fmt.Sprintf("Cannot parse precision `%s`: %s", values[0], err.Error()),
+		}
+	}
+
+	scale, err := strconv.ParseInt(values[1], 10, 32)
+	if err != nil {
+		return 0, 0, adbc.Error{
+			Code: adbc.StatusInvalidData,
+			Msg:  fmt.Sprintf("Cannot parse scale `%s`: %s", values[1], err.Error()),
+		}
+	}
+	return int32(precision), int32(scale), nil
+}
+
+func patternToRegexp(pattern *string) (*regexp.Regexp, error) {
+	patternString := ""
+	if pattern != nil {
+		patternString = *pattern
+	}
+	patternString = strings.TrimSpace(patternString)
+
+	convertedPattern := ".*"
+	if patternString != "" {
+		convertedPattern = fmt.Sprintf("(?i)^%s$", strings.ReplaceAll(strings.ReplaceAll(patternString, "_", "."), "%", ".*"))
+	}
+	r, err := regexp.Compile(convertedPattern)
+	if err != nil {
+		return nil, adbc.Error{
+			Code: adbc.StatusInvalidArgument,
+			Msg:  fmt.Sprintf("Cannot parse pattern `%s`: %s", patternString, err.Error()),
+		}
+	}
+	return r, nil
+}
```

Comment:
Does bigquery not support SQL pattern syntax?

Reply:
Sorry this was found in the csharp codebase and I thought that it was there for some careful thoughts and decisions so I copied the implementation. I'll remove this function.

https://github.com/apache/arrow-adbc/blob/282819d7e8e6e3cfcaad0d15fd8cd4bfa1502279/csharp/src/Drivers/BigQuery/BigQueryConnection.cs#L735-L746